src/v/random/generators.cc

```diff
@@ -32,4 +32,18 @@ ss::sstring gen_alphanum_string(size_t n) {
     return s;
 }
 
+ss::sstring gen_alphanum_max_distinct(size_t cardinality) {
+    static constexpr std::size_t num_chars = chars.size() - 1;
+    // everything is deterministic once you choose key_num
+    auto key_num = get_int(cardinality - 1);
+    auto next_index = key_num % num_chars;
+    auto s = ss::uninitialized_string(alphanum_max_distinct_strlen);
+    std::generate_n(s.begin(), alphanum_max_distinct_strlen, [&] {
+        auto c = chars[next_index];
+        next_index = (next_index + key_num) % num_chars;
```

Comment:
seems to me this is high _non_-random. it seems like it might be better to choose random() % cardinality so that you are selecting randomly within a range whose size is constrained by cardinatility.

Reply:
My goal is to choose k arbitrary keys.

Reply:
Also feel free to share a code snippet. Not sure the details of what you are proposing other than keys in [0,n) with the same distribution but integral values.

Reply:
what i'm saying is that in your test if you want key cardinality 100 then you could use `fmt::format("{}", get_int(100));`

Reply:
It's a good point. I can do a larger fmt template for bigger keys. I was originally going to make max len a variable too, but just wanted a free function that acted like a generator, and having to use the same max size and max cardinality args each time to get the same set felt awkward.